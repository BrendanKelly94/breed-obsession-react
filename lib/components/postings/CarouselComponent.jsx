import React from 'react';
import { UncontrolledCarousel } from 'reactstrap';
const CarouselComponent = ({ pics }) => {
  let slides;
  if(pics){
    slides = pics.map((pic) => {
      return {
        src: pic['$t'],
      };
    });
  }
  return (
    <UncontrolledCarousel items = {slides} autoPlay = {false} />
  );
};
export default CarouselComponent;
