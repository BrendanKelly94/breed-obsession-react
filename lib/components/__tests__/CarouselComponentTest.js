import React from 'react';
import CarouselComponent from '../postings/CarouselComponent.jsx';

import { shallow } from 'enzyme';

describe('CarouselComponent', () => {

  const testProps = {
    pics: []
  };

  it('renders correctly', () => {
    const wrapper = shallow(
      <CarouselComponent
        {...testProps}
      />
    );

    expect(wrapper).toMatchSnapshot();

  });
});
