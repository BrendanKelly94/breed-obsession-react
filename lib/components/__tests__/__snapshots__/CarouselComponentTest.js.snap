// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselComponent renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <UncontrolledCarousel
    autoPlay={false}
    controls={true}
    indicators={true}
    items={Array []}
  />,
  "nodes": Array [
    <UncontrolledCarousel
      autoPlay={false}
      controls={true}
      indicators={true}
      items={Array []}
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <CarouselComponent
      pics={Array []}
    />,
    "_forcedUpdate": false,
    "_instance": null,
    "_newState": null,
    "_rendered": <UncontrolledCarousel
      autoPlay={false}
      controls={true}
      indicators={true}
      items={Array []}
    />,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CarouselComponent
    pics={Array []}
  />,
}
`;
